src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -2171,7 +2171,7 @@ self =>
         }
         def checkWildStar: Tree = top match {
           case Ident(nme.WILDCARD) if isSequenceOK && isRawStar => peekingAhead (
-            if (isCloseDelim) atPos(top.pos.start, in.prev.offset)(Star(stripParens(top)))
+            if (isCloseDelim) atPos(top.pos.start, in.prev.offset)(Star(top))
```

Comment:
Why was stripParens removed here?

Reply:
law of conservation of stripParens. Also, top was just matched on Ident so no parens. Then where did the parens go?

I'm about to push another commit with minor refactor.

Reply:
Oh right, I think I saw this debated on dotty:
```
scala> List(42) match { case Seq((xs)*) => xs }
1 |List(42) match { case Seq((xs)*) => xs }
  |                               ^
  |                               `*` must follow pattern variable
```
even though parens look innocent here.

Reply:
Scala 2 is ok with
```
List(42) match { case Seq((xs) @ _*) => xs }
```
but not
```
scala> List(42) match { case Seq((xs)*) => xs }
                                      ^
       error: bad simple pattern: use _* to match a sequence
```
under `-Xsource:3`